### PR TITLE
[V4] Cleanup and simplify ocean.util.log.Config

### PR DESCRIPTION
**Pull request by mathias-lang-sociomantic**
The first 3 commits are pretty straightforward and remove some oddities/indirections introduced with the duplication of the loggers.

The last one is something that bugged me when I was working on this code. Having a mix of template arguments and runtime lookup is not very consistent. Using the `makeLayout` allows one to implement whatever default they wish while simplifying the interface for everyone.

### CODE REVIEW

**Comment by mathias-lang-sociomantic** [comment]
Ping

**Review by mihails-strasuns-sociomantic — APPROVED** [review]
I just checked and looks like last commit won't affect any of our projects.

File: relnotes/Logger-Config.migration.md [hunk 1]
```diff
@@ -0,0 +1,19 @@
+# Removed legacy template arguments in Logger's configuration module
```
Comment (leandro-lucarella-sociomantic) [inline relnotes/Logger-Config.migration.md:1]: This format is incorrect, it should be `##`.

File: relnotes/Logger-Config.migration.md [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+* `ocean.util.log.Config : setupLoggerLevel, configureLogger`
```
Comment (leandro-lucarella-sociomantic) [inline relnotes/Logger-Config.migration.md:3]: And there should be no `*`.

File: relnotes/Logger-Config.migration.md [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+
+Those functions used to take a `LoggerT` template parameter to signify which logger to configure.
+As the old logger implementation is now removed, this parameter is now gone.
+No disturbance is expected in user's code, as it's unlikely that those parameter were provided explicitly,
+since the template parameter was tied to one of the runtime parameter.
+
+# Config-based default for console_layout and file_layout
+
+* `ocean.util.log.Config : Config.{console_layout,file_layout}, configureLoggers`
```
Comment (leandro-lucarella-sociomantic) [inline relnotes/Logger-Config.migration.md:12]: The same for this entry.

**Pull request merged**